tests/jballerina-unit-test/src/test/resources/test-src/object/distinct_objects_negative.bal

```diff
@@ -48,3 +54,24 @@ function foo() {
     g(o1);
     g(o2);
 }
+
+public function testDistinctObjectAssignability() {
+    object {} x = object {};
+    Foo _ = x;
+}
+
+public function testDistinctObjectAssignability2() {
+    object {
+        int i;
+    } x = object {
+        int i = 1;
+    };
+
+    Bar _ = x;
+
+    x = object {
+        int i = 10;
+    };
+
+    Bar _ = x;
+}
```

Comment:
When adding negative tests, let's mark the place where the error occurs(Simply we can add a comment (// error) in the relevant line). Please check other negative test files

Reply:
I will add

Reply:
Updated